continuous_integration/flake8_diff.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+
+set -e
+
+# Travis does the git clone with a limited depth (50 at the time of
+# writing). This may not be enough to find the common ancestor with
+# $REMOTE/master so we unshallow the git checkout
+git fetch --unshallow || echo "Unshallowing the git checkout failed"
```

Comment:
For completeness here are some very rough timings compared for the time `git clone --depth 50` and `git fetch --unshallow`:

              | git clone depth 50 | git unshallowing |
--------------|--------------------|------------------  |
 scikit-learn | 10s                | 40s              |
 joblib       | 1-2s               | 2-3s             |